docs/connector-development/config-based/tutorial/3-connecting-to-the-API-source.md

```diff
@@ -0,0 +1,238 @@
+  stream_names: ["customers_stream"]
+
+```
+
+Let's fill this out these TODOs with the information found in the [Exchange Rates API docs](https://exchangeratesapi.io/documentation/)
+
+1. First, let's rename the stream from `customers` to `rates`, and update the primary key to `date`
+
+```
+rates_stream:
+  type: DeclarativeStream
+  $options:
+    name: "rates"
+    primary_key: "date"
+```
+
+and update the references in the streams list and check block
+
+```
+streams:
+  - "*ref(rates_stream)"
+check:
+  type: CheckStream
+  stream_names: ["rates"]
+```
+
+2. Next we'll set the base url.
+   According to the API documentation, the base url is `"https://api.exchangeratesapi.io/v1/"`.
+   This can be set in the requester definition.
+
+```
+requester:
+  type: HttpRequester
+  name: "{{ options['name'] }}"
+  url_base: "https://api.exchangeratesapi.io/v1/" # Only change the url_base field
+```
+
+3. We can fetch the latest data by submitting a request to "/latest". This path is specific to the stream, so we'll set within the `rates_stream` definition.
+
+```
+rates_stream:
+  type: DeclarativeStream
+  $options:
+    name: "rates"
+    primary_key: "date"
+  schema_loader:
+    $ref: "*ref(schema_loader)"
+  retriever:
+    $ref: "*ref(retriever)"
+    requester:
+      $ref: "*ref(requester)"
+      path: "/latest"
+```
+
+4. Next, we'll set up the authentication.
+   The Exchange Rates API requires an access key to be passed as request parameter. We'll need to make this access key accessible to our connector, and pass it as a request_parameter in the `request_parameters` field of the `request_options_provider`
+   We'll configure the connector to use this access key by setting the access key in a request parameter and pointing to a field in the config, which we'll populate in the next step:
+
+```
+requester:
+  type: HttpRequester
+  name: "{{ options['name'] }}"
+  url_base: "https://api.exchangeratesapi.io/v1/"
+  http_method: "GET"
+  request_options_provider:
+    request_parameters:
+      access_key: "{{ config.access_key }}"
+```
+
+Since the access key is set directly as a request parameter, we can remove the `authentication` field from the `requester`.
+
+5. According to the ExchangeRatesApi documentation, we can specify the base currency of interest in a request parameter:
+
+```
+request_options_provider:
+  request_parameters:
+    access_key: "{{ config.access_key }}"
+    base: "{{ config.base }}"
+```
+
+The full connection definition should now look like
+
+```
+schema_loader:
+  type: JsonSchema
+  file_path: "./source_exchange_rates_tutorial/schemas/{{ options.name }}.json"
+selector:
+  type: RecordSelector
+  extractor:
+    type: JelloExtractor
+    transform: "_"
+requester:
+  type: HttpRequester
+  name: "{{ options['name'] }}"
+  url_base: "https://api.exchangeratesapi.io/v1/"
+  http_method: "GET"
+  request_options_provider:
+    request_parameters:
+      access_key: "{{ config.access_key }}"
```

Comment:
idk if they recently changed this but this now goes into the headers as `apikey`. see https://apilayer.com/marketplace/exchangerates_data-api#documentation-tab

Reply:
I used this as reference https://exchangeratesapi.io/documentation/
I'll switch to a header if it works because we have a builtin authenticator

Reply:
the demo is hitting a different endpoint https://api.apilayer.com vs https://api.exchangeratesapi.io/v1/.
I'll update to https://api.apilayer.com because it seems like the newer version and it uses header-based authentication